presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/IPrestoSparkTaskCompiler.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.util.CollectionAccumulator;
+import scala.Tuple2;
+
+import java.util.Iterator;
+import java.util.Map;
+
+public interface IPrestoSparkTaskCompiler
+{
+    Iterator<Tuple2<Integer, byte[]>> compile(
```

Comment:
I am wondering if this should be called `IPrestoSparkTaskExecutor` and the method name is `execute`? -- basically this did local execution plan and actually execute it :)

Reply:
> I am wondering if this should be called IPrestoSparkTaskExecutor

The `IPrestoSparkTaskCompiler` does not execute the task (it does not process any data), it only creates a processor. Thus it feels like `IPrestoSparkTaskExecutor` would be missleading.  The actually execution happens in the Iterator itself (when next() is called).